### DISCUSSION TITLE
Open-Source Code

### DISCUSSION THREAD

**Discussion by Gurkengewuerz** [rg35xx-cfw/Koriki#14] [Q&A] [answered]
Hey @acmeplus,

thanks a lot for this work. As a developer and Linux enthusiast, it's really cool to be part of the development of such projects.
I'm just wondering that the code in the repository has not been updated for a while? Isn't it the plan to keep the code up to date, which I would find quite a pity, because this also took no positive direction with GarlicOS.
Also, I would like to be able to create PR and help configure Retroarch or Drastic.

Best regards from germany. ✌🏼

**Comment by tsipizic** [comment]
I would also like to help with the development, I can image the same is true for many others. An up to date repository and standard open source development practices would help the project greatly imho.

**Reply by acmeplus** [reply]
Curious what you mean with "standard open source development practices"? The project is open for anyone to collaborate, issues are there and some folks collaborate with configuration changes, but so far no one has produced a single code PR, and as I mentioned in the previous answer there are no updates recently because of work, other projects, and in general vacation time :)

**Reply by tsipizic** [reply]
I was not referring to anything specific other than branches, PRs etc. For example a user mentioned a Mupen64 configuration change in an issue but this is not committed to the main repository as it is not up to date with your other changes. As long as I am concerned I am satisfied with your intent of keeping the code open source in the long term. Thanks for your reply and your time spent on the project.

**Reply by acmeplus** [reply]
Yeah, that's a good example where quick changes could be applied. The only problem is right now the build is not in a real CI/CD environment but it's manually rebuilt on my machine, sometimes I have temporary changes that I'm testing and I don't want to have several full builds of the buildroot firmware in parallel. One possibility is to split the actual configuration files from the main build, and distribute configuration changes independently. In any case that may be something that could be added after 1.0. Right now I bought another rg35xx to replicate the v6C issues, and I need to consolidate the system to it works on all models transparently. Once that's done I'll continue to close all the pending issues for 1.0.

**Comment by acmeplus** [comment, marked as answer]
Almost of the code is already published in github. There have been many updates recently because I've been busy with work, other projects, and of course vacation... but in general the code is kept more or less up to date. There are areas where anyone can help without the need to have an up to date copy, and is in all the aspects related to the simplemenu configuration, launchers, cores, etc. 90% of the problems that users report are based on incorrect configurations, or missing elements.

For the actual low level buildroot stuff, the first step for anyone to help is to get the system compiling and get their own buildroot built. There may be areas that don't work with a single make compilation, but report back and will tell you what needs to change. Note that I don't recompile the full buildroot all the time, so I may not be aware of all the problems.

Then there are portions of the kernel that are not public because those were disclosed by the vendor under the condition of not re-distributing them. The kernel published in github is 90% there and mostly functional, in fact all the initial releases were based on that kernel.

But in general, yes, help is appreciated and the intention of this project is to continue being open source. So get your build system together and let me know if you need help to progress. Then we can start distributing the opened issues, etc. 

Thanks!

**Reply by hormigas** [reply]
I appreciate your work and candidness. But you're also saying that some of the kernel source you won't be releasing at any point? Isn't it also GPL'd?
